Tensile/SolutionStructs.py

```diff
@@ -4945,20 +4947,27 @@ def getParameterValueAbbreviation( key, value ):
   ##########################
   # make class look like dict
   def keys(self):
-    return list(self._state.keys())
+    mkeys = self._mutable_state.keys()
+    ikeys = list(self._state.keys())
+    return ikeys + list(mkeys) if mkeys else ikeys
 
   def __len__(self):
-    return len(self._state)
+    return len(self._state) + len(self._mutable_state)
 
   def __iter__(self):
-    return iter(self._state)
+    temp = self._mutable_state
+    temp.update(self._state)
+    return iter(temp)
 
   def __getitem__(self, key):
+    if key in self._mutable_state:
+      return self._mutable_state[key]
     return self._state[key]
 
   def __setitem__(self, key, value):
+    printWarning(f"Mutating solution: {self._name} {key}")
     self._name = None
-    self._state[key] = value
+    self._mutable_state[key] = value
 
```

Comment:
What if a key that already exists on `_state` is added via __set_item__? Should we reject over-writing immutable state with mutable state?

Similarly, if a key exists in both `_state` and `_mutable_state`, __getitem__ will prefer the mutable option. Is this the behaviour we're after in all cases?

Reply:
The mutated state should be considered the "truth" at least that is my intention. Probably need to clean up a few of the functions to be sure that this is enforced.

Reply:
This impl was abandoned